### PR TITLE
Support other casings for `view-controller`

### PR DESCRIPTION
**Pull request by seanpdoyle**
The decision to use dash-casing for the `view-controller` property matches the multi-word precedent set by the [Turbo iOS examples][]. However, the [Turbo Android examples][] uses snake_casing.

It's unlikely that an Android app and an iOS app will share exact copies of a path configuration file's structure and data. However, it's possible that they'd share a similar server-side backend to generate those files, or a shared naming convention across those files.

The `view-controller` appears to be the only multi-word key expected and supported out-of-the-box, with the expectation that other keys can be used to inform bespoke features in app-specific contexts.

This commit expands that support to match other naming conventions, mainly snake_case and camelCase.

[Turbo iOS examples]: https://github.com/hotwired/turbo-ios/blob/c9d060fcdec136c3808cd0764e956d375012c6fb/Docs/PathConfiguration.md#path-configuration
[Turbo Android examples]: https://github.com/hotwired/turbo-android/blob/bf945e18aed3d5749c9efc6e228d98bc012878af/docs/PATH-CONFIGURATION.md#path-configuration

### CODE REVIEW

File: Sources/VisitProposalExtension.swift [hunk 1]
```diff
@@ -39,10 +39,8 @@ public extension VisitProposal {
     ///
     /// A default value is provided in case the view controller property is missing from the configuration file. This will route the default `VisitableViewController`.
     var viewController: String {
-        if let viewController = properties["view-controller"] as? String {
-            return viewController
-        }
+        let viewControllers = ["view-controller", "view_controller", "viewController"].map { properties[$0] }.filter { $0 as? String }
```
Comment (seanpdoyle) [inline Sources/VisitProposalExtension.swift:42]: I am an amateur Swift developer reaching for Ruby-isms. If there's a more idiomatic way to do this, I'd love to learn it!

File: Sources/VisitProposalExtension.swift [hunk 2]
```diff
@@ -39,10 +39,8 @@ public extension VisitProposal {
...
 
-        return VisitableViewController.pathConfigurationIdentifier
+        return viewControllers.first || VisitableViewController.pathConfigurationIdentifier
```
Comment (joemasilotti) [inline Sources/VisitProposalExtension.swift:44]: What's the purpose of this change?

**Review by joemasilotti — COMMENTED** [review]
Great idea! Other options are coming so it's a great idea to have this in place now before those are added.

There might be a more "Swift-y" way to do this, I'm going to dig in.

**Comment by joemasilotti** [comment]
Hey Sean, after discussing with the other maintainers we've decided to consolidate case. **Everything, across both platforms, will use snake case moving forward.** This better matches Rails routes and requires the least amount of change across iOS and Android.

You can check out the [turbo-navigator branch](https://github.com/hotwired/turbo-ios/pull/158) if you'd like to see how everything is handled and take it for a spin. I'm hoping to merge that in pretty soon!

Thanks for the PR, it kicked off a good discussion. And I think we are in a much better spot because of it.
